src/main/java/io/jenkins/pluginhealth/scoring/model/Plugin.java

```diff
@@ -89,6 +100,15 @@ public Plugin setReleaseTimestamp(ZonedDateTime releaseTimestamp) {
         return this;
     }
 
+    public Map<String, String> getDetails() {
+        return new HashMap<>(details);
```

Comment:
Quoting from [here](https://stackoverflow.com/a/65659020): Regarding `Map.copyOf()`,

`But this has a price: with a copy, you need two times the amount of memory for the map (roughly). For really large maps, this may cause memory shortages up to an OutOfMemoryError`

Is this going to be a problem for us in the future?

Reply:
we can deal with this when it happens